File: tests/fixtures/source_files/calculator.js
/**
 * Simple JavaScript calculator for testing
 */
class Calculator {
    constructor() {
        this.result = 0;
    }

    add(value) {
        this.result += value;
        return this.result;
    }

    subtract(value) {
        this.result -= value;
        return this.result;
    }

    multiply(value) {
        this.result *= value;
        return this.result;
    }

    divide(value) {
        if (value === 0) {
            throw new Error("Division by zero");
        }
        this.result /= value;
        return this.result;
    }

    getResult() {
        return this.result;
    }

    clear() {
        this.result = 0;
    }
}

function calculateArea(length, width) {
    if (length <= 0 || width <= 0) {
        throw new Error("Dimensions must be positive");
    }
    return length * width;
}

module.exports = { Calculator, calculateArea };

File: tests/fixtures/source_files/calculator.rs
// Simple Rust calculator for testing
pub struct Calculator {
    result: f64,
}

impl Calculator {
    pub fn new() -> Self {
        Calculator { result: 0.0 }
    }

    pub fn add(&mut self, value: f64) -> f64 {
        self.result += value;
        self.result
    }

    pub fn subtract(&mut self, value: f64) -> f64 {
        self.result -= value;
        self.result
    }

    pub fn multiply(&mut self, value: f64) -> f64 {
        self.result *= value;
        self.result
    }

    pub fn divide(&mut self, value: f64) -> Result<f64, String> {
        if value == 0.0 {
            Err("Division by zero".to_string())
        } else {
            self.result /= value;
            Ok(self.result)
        }
    }

    pub fn get_result(&self) -> f64 {
        self.result
    }

    pub fn clear(&mut self) {
        self.result = 0.0;
    }
}